src/refmvs.rs

```diff
@@ -486,115 +486,76 @@ unsafe extern "C" fn add_temporal_candidate(
         }
     };
 }
-unsafe extern "C" fn add_compound_extended_candidate(
-    same: *mut refmvs_candidate,
-    same_count: *mut libc::c_int,
-    cand_b: *const refmvs_block,
-    sign0: libc::c_int,
-    sign1: libc::c_int,
+
+fn add_compound_extended_candidate(
+    same: &mut [refmvs_candidate],
+    same_count: &mut [usize; 4],
+    cand_b: &refmvs_block,
+    sign0: u8,
+    sign1: u8,
     r#ref: refmvs_refpair,
-    sign_bias: *const uint8_t,
+    sign_bias: &[u8; 7],
 ) {
-    let diff: *mut refmvs_candidate = &mut *same.offset(2 as libc::c_int as isize)
-        as *mut refmvs_candidate;
-    let diff_count: *mut libc::c_int = &mut *same_count.offset(2 as libc::c_int as isize)
-        as *mut libc::c_int;
-    let mut n: libc::c_int = 0 as libc::c_int;
-    while n < 2 as libc::c_int {
-        let cand_ref: libc::c_int = (*cand_b).r#ref.r#ref[n as usize] as libc::c_int;
-        if cand_ref <= 0 as libc::c_int {
+    let (same, diff) = same.split_at_mut(2);
+
+    // Alternative way with compile-time index checking
+    // for the resulting array refs (rather than slices).
+    //
+    // Safety: `[usize; 4]` and `[[usize; 2]; 2]` should be identical.
+    // let [same_count, diff_count] = unsafe { 
+    //     std::mem::transmute::<&mut [usize; 4], &mut [[usize; 2]; 2]>(same_count)
+    // };
```

Comment:
Do we need this comment? Seems like it's documenting a different approach that you're no longer using?

Reply:
Yeah, I added it in case we want to implement it that way (trading a transmute for compile-time indexing).  Should I remove it?

Reply:
I think that's an approach that would be cool to take once safe transmute becomes a thing, but probably not worth the extra unsafe operation when `split_at` already does the trick nicely. So I'd say remove the comment here.

Reply:
Removed in 5ea222199003cae8abd46c2b7236c3e11859c067.